alacritty_terminal/src/config/mod.rs

```diff
@@ -143,7 +143,32 @@ pub struct Config<T> {
     pub persistent_logging: Option<bool>,
 }
 
+#[derive(Debug, PartialEq)]
+pub struct ConfigValidationError {
+    field: String,
+    message: String,
+}
+
+impl ::std::error::Error for ConfigValidationError {
+    fn description(&self) -> &str {
+        self.message.as_str()
+    }
+}
+
+impl Display for ConfigValidationError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        Display::fmt(&self.message, f)
+    }
+}
+
 impl<T> Config<T> {
+    pub fn validate(&self) -> Option<ConfigValidationError> {
+        self.working_directory.as_ref().filter(|x| !x.exists()).map(|_| ConfigValidationError {
+            field: String::from("working_directory"),
+            message: String::from("working directory does not exist"),
```

Comment:
FWIW, this is our current handling for the config file, maybe this could be similar:

https://github.com/jwilm/alacritty/blob/master/alacritty/src/config/mod.rs#L150